core/src/main/java/tachyon/client/TachyonFile.java

```diff
@@ -204,10 +204,13 @@ public String getLocalFilename(int blockIndex) throws IOException {
    * @throws IOException
    */
   public List<String> getLocationHosts() throws IOException {
+    List<String> ret = new ArrayList<String>();
+    if (getNumberOfBlocks() == 0) {
+      return ret;
+    }
```

Comment:
I don't think this block is necessary if we initialize the return list as an empty list since it will just skip the next if statement and return unaltered.